File: Java/Product.java
public class Product {
    // Common attributes for all products
    private int id;        // unique identifier
    private String name;   // product name
    private String brand;  // product brand
    private int price;     // product price

    // Default constructor
    public Product() {
        this.id = 0;
        this.name = "";
        this.brand = "";
        this.price = 0;
    }

    // Constructor with parameters
    public Product(int id, String name, String brand, int price) {
        this.id = id;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    // Getters
    public int getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }

    public String getBrand() {
        return this.brand;
    }

    public int getPrice() {
        return this.price;
    }

    // Setters
    public void setName(String name) {
        this.name = name;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}

File: Python/Product.py
class Product:
    def __init__(self, id=0, name="", brand="", price=0):
        # Private attributes for product details
        self.__id = id          # Unique product ID
        self.__name = name      # Product name
        self.__brand = brand    # Product brand
        self.__price = price    # Product price

    # Getters
    def get_id(self):
        return self.__id

    def get_name(self):
        return self.__name

    def get_brand(self):
        return self.__brand

    def get_price(self):
        return self.__price

    # Setters
    def set_name(self, name):
        self.__name = name

    def set_brand(self, brand):
        self.__brand = brand

    def set_price(self, price):
        self.__price = price
